cmd/vspd/main.go

```diff
@@ -128,14 +128,16 @@ func run() int {
 	// Periodically write a database backup file.
 	shutdownWg.Add(1)
 	go func() {
-		select {
-		case <-ctx.Done():
-			shutdownWg.Done()
-			return
-		case <-time.After(cfg.BackupInterval):
-			err := db.WriteHotBackupFile()
-			if err != nil {
-				log.Errorf("Failed to write database backup: %v", err)
+		for {
```

Comment:
I saw this in the original PR, but thought it was intended to only be a single backup after the interval had expired.  I should've asked about it.

Reply:
As it was it created a time-bomb. After the backup interval had passed it became impossible to cleanly shutdown the process.

Reply:
Yeah, makes sense.  It would indeed do that because `shutdownWg.Done()` isn't deferred so it wouldn't run.